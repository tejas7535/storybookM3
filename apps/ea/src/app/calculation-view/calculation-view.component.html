<schaeffler-breadcrumbs
  *ngIf="isStandalone$ | ngrxPush"
  class="mx-4 mt-4 hidden sm:block"
  [breadcrumbs]="breadcrumbs$ | ngrxPush"
></schaeffler-breadcrumbs>
<div class="flex flex-1 max-w-full justify-center">
  <ea-calculation-container></ea-calculation-container>
</div>
