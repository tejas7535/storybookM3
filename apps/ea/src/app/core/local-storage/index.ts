export * from './ea-parameters-local-storage-item.model';
export * from './local-storage.service';
