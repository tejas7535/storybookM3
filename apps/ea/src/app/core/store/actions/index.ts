export * from './calculation-parameters';
export * from './calculation-result';
export * from './product-selection';
export * from './settings';
export * from './storage-messages';
