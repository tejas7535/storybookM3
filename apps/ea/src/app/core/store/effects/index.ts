export * from './calculation-parameters';
export * from './calculation-result';
