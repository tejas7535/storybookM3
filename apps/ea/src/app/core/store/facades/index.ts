export * from './calculation-parameters/calculation-parameters.facade';
export * from './settings/settings.facade';
