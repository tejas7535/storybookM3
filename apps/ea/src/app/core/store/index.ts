export * from './facades';
export * from './reducers';
export * from './selectors';
export * from './store.module';
