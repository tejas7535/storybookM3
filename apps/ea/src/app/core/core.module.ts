import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import {
  ErrorStateMatcher,
  ShowOnDirtyErrorStateMatcher,
} from '@angular/material/core';
import { MatIconRegistry } from '@angular/material/icon';
import { DomSanitizer } from '@angular/platform-browser';

import { environment } from '@ea/environments/environment';
import {
  AVAILABLE_LANGUAGES,
  FALLBACK_LANGUAGE,
  LANGUAGE_STORAGE_KEY,
} from '@ea/shared/constants/language';
import {
  TRANSLOCO_PERSIST_LANG_STORAGE,
  TranslocoPersistLangModule,
} from '@ngneat/transloco-persist-lang';

import { SharedTranslocoModule } from '@schaeffler/transloco';

import { HttpBearinxInterceptor } from './interceptor/http-bearinx.interceptor';
import { HttpCatalogWebApiInterceptor } from './interceptor/http-catalog-web-api.interceptor';
import { HttpCO2UpstreamInterceptor } from './interceptor/http-co2-upstream.interceptor';
import { StoreModule } from './store/store.module';

@NgModule({
  imports: [
    StoreModule,

    HttpClientModule,
    SharedTranslocoModule.forRoot(
      environment.production,
      AVAILABLE_LANGUAGES,
      undefined, // default -> undefined would lead to browser detection
      FALLBACK_LANGUAGE.id,
      LANGUAGE_STORAGE_KEY,
      true,
      !environment.localDev,
      undefined,
      `${environment.assetsPath}/i18n/`
    ),

    TranslocoPersistLangModule.forRoot({
      storageKey: LANGUAGE_STORAGE_KEY,
      storage: {
        provide: TRANSLOCO_PERSIST_LANG_STORAGE,
        useValue: localStorage,
      },
    }),
  ],

  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpBearinxInterceptor,
      multi: true,
    },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpCatalogWebApiInterceptor,
      multi: true,
    },
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpCO2UpstreamInterceptor,
      multi: true,
    },
    { provide: ErrorStateMatcher, useClass: ShowOnDirtyErrorStateMatcher },
  ],

  exports: [StoreModule, SharedTranslocoModule],
})
export class CoreModule {
  public constructor(
    private readonly matIconRegistry: MatIconRegistry,
    private readonly sanitizer: DomSanitizer
  ) {
    this.registerEAIcons();
  }

  public registerEAIcons(): void {
    const iconSet: Record<string, string> = {
      co2: 'icon_CO2.svg',
      airwaves: 'icon_airwaves.svg',
      air: 'icon_air.svg',
      acute: 'icon_acute.svg',
      bolt: 'icon_bolt.svg',
      device_thermostat: 'icon_device_thermostat.svg',
      mop: 'icon_mop.svg',
      calculation: 'icon_calculations.svg',
      friction_load: 'icon_load_frictional_powerloss.svg',
      lubrication_parameters: 'icon_lubrication_parameters.svg',
      rating_life: 'icon_rpm_rating_life.svg',
      water_drop: 'icon_water_drop.svg',
    };
    for (const [name, url] of Object.entries(iconSet)) {
      const setUrl = this.sanitizer.bypassSecurityTrustResourceUrl(
        `${environment.assetsPath}/icons/${url}`
      );
      this.matIconRegistry.addSvgIcon(name, setUrl);
    }
  }
}
