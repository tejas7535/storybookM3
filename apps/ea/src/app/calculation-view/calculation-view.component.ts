import { NgIf } from '@angular/common';
import { Component } from '@angular/core';

import { BehaviorSubject } from 'rxjs';

import { AppRoutePath } from '@ea/app-route-path.enum';
import { CalculationContainerComponent } from '@ea/calculation/calculation-container/calculation-container.component';
import { SettingsFacade } from '@ea/core/store';
import { TranslocoService } from '@ngneat/transloco';
import { TranslocoDecimalPipe } from '@ngneat/transloco-locale';
import { PushPipe } from '@ngrx/component';

import { Breadcrumb, BreadcrumbsModule } from '@schaeffler/breadcrumbs';

@Component({
  selector: 'ea-calculation-view',
  templateUrl: './calculation-view.component.html',
  standalone: true,
  imports: [CalculationContainerComponent, BreadcrumbsModule, PushPipe, NgIf],
  providers: [TranslocoDecimalPipe],
})
export class CalculationViewComponent {
  public readonly isStandalone$ = this.settingsFacade.isStandalone$;

  public breadcrumbs$: BehaviorSubject<Breadcrumb[]> = new BehaviorSubject(
    this.getBreadcrumbs()
  );

  constructor(
    private readonly translocoService: TranslocoService,
    private readonly settingsFacade: SettingsFacade
  ) {}

  private getBreadcrumbs(): Breadcrumb[] {
    return [
      {
        label: this.translocoService.translate('breadcrumbs.landingPage'),
        url: AppRoutePath.HomePath,
      },
      {
        label: this.translocoService.translate('breadcrumbs.calculation'),
      },
    ];
  }
}
