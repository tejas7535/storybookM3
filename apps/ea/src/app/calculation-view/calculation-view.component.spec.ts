import { MatDividerModule } from '@angular/material/divider';
import { MatIconTestingModule } from '@angular/material/icon/testing';
import { ActivatedRoute } from '@angular/router';

import { BehaviorSubject, of } from 'rxjs';

import { SettingsFacade } from '@ea/core/store';
import { APP_STATE_MOCK } from '@ea/testing/mocks';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { provideMockStore } from '@ngrx/store/testing';
import { MockModule } from 'ng-mocks';

import {
  BreadcrumbsComponent,
  BreadcrumbsModule,
} from '@schaeffler/breadcrumbs';
import { provideTranslocoTestingModule } from '@schaeffler/transloco/testing';

import { CalculationViewComponent } from './calculation-view.component';

describe('CalculationViewComponent', () => {
  let component = CalculationViewComponent;
  let spectator: Spectator<CalculationViewComponent>;

  const standaloneChanges$ = new BehaviorSubject<boolean>(true);
  const standalone$ = standaloneChanges$.asObservable();

  const createComponent = createComponentFactory({
    component: CalculationViewComponent,
    imports: [
      provideTranslocoTestingModule({ en: {} }),
      MatIconTestingModule,
      MockModule(MatDividerModule),
      MockModule(BreadcrumbsModule),
    ],
    providers: [
      provideMockStore({
        initialState: {
          ...APP_STATE_MOCK,
        },
      }),
      {
        provide: ActivatedRoute,
        useValue: {
          paramMap: of(new Map()),
        },
      },
      {
        provide: SettingsFacade,
        useValue: {
          isStandalone$: standalone$,
        },
      },
    ],
    detectChanges: false,
  });

  beforeEach(async () => {
    spectator = createComponent();
    component = spectator.debugElement.componentInstance;
    spectator.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should have breadcrumbs in the template', () => {
    spectator.detectChanges();
    const breadcrumbs = spectator.query(BreadcrumbsComponent);
    expect(breadcrumbs).toBeTruthy();
    expect(breadcrumbs.breadcrumbs).toMatchSnapshot();
  });

  describe('when not a standalone version', () => {
    beforeEach(() => {
      standaloneChanges$.next(false);
      spectator.detectChanges();
    });

    it('should not have breadcrumbs in the template', () => {
      const breadcrumbs = spectator.query(BreadcrumbsComponent);
      expect(breadcrumbs).toBeFalsy();
    });
  });
});
