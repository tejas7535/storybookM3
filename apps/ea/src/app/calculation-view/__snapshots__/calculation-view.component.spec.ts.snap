// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculationViewComponent should have breadcrumbs in the template 1`] = `
Array [
  Object {
    "label": "breadcrumbs.landingPage",
    "url": "home",
  },
  Object {
    "label": "breadcrumbs.calculation",
  },
]
`;
